### DISCUSSION TITLE
Colors of Demand and Gantt Report

### DISCUSSION THREAD

**Discussion by zipus** [frePPLe/frepple#569] [Q&A]
On Demand, there are colors (yellow, orange, red) if the demand is delayed. Which is the limit/meaning for each color?
I found also that de demand gantt report bars shows colors (yellow, grey, and [purple also in the docs](https://frepple.com/docs/current/user-interface/plan-analysis/demand-gantt-report.html)).
Which are the meaning for this colors?
Can we also add the meaning to docs?

**Comment by jdetaeye** [comment]
> On Demand, there are colors (yellow, orange, red) if the demand is delayed. Which is the limit/meaning for each color?

On demand screen it's only red and green:
![image](https://user-images.githubusercontent.com/5219207/199496221-9dd55711-828a-4af0-a977-aaf95ea3e2d3.png)

> I found also that de demand gantt report bars shows colors (yellow, grey, and [purple also in the docs]
> Which are the meaning for this colors?

In the Community Edition these colors are purely based on the delay of the manufacturing orders, purchase orders and distribution orders.  Ie whether the operation is planned early enough to allow all demand(s) it feeds to be on time.

In the Enterprise&Cloud Editions, there is a more advanced concept for coloring. For make-to-stock items, the MO/PO/DO then can get a color that represents the stock level of the buffer they are feeding.  This is documented at https://frepple.com/docs/current/user-interface/plan-analysis/inventory-status.html

In the "plan editor" (also only in the Enterprise&Cloud Editions) users have much more flexibility to configure their own colors and limits: 
https://frepple.com/docs/current/user-interface/plan-analysis/plan-editor.html

**Reply by zipus** [reply]
I'm talking about these colors, not the Enterprise plan editor:
![imagen](https://user-images.githubusercontent.com/1032986/199721948-5bfa8884-94e7-4155-af69-8786d7d65d7b.png)
Which are the possible color values and their meaning? The docs only talk about the vertical black and red lines and not the yellow, grey, purple bars.

> In the "plan editor" (also only in the Enterprise&Cloud Editions) users have much more flexibility to configure their own colors and limits: https://frepple.com/docs/current/user-interface/plan-analysis/plan-editor.html

> ....shows colors (yellow, grey, and [purple also in the docs](https://frepple.com/docs/current/user-interface/plan-analysis/demand-gantt-report.html)).

**Comment by zipus** [comment]
I have more than 2 colors in Demand 6.25.0 Community Edition:
![image](https://user-images.githubusercontent.com/1032986/199552893-820d1403-0959-4aa1-aec7-ee7b86f713ef.png)

**Reply by jdetaeye** [reply]
That color is pretty simple:   Everything delayed for more than 7 days is red.  Between 0 and 7 days lateness the color gradually changes from yellow to red.   And, yes, the 7 day limit is hardcoded.
